includes/admin_import.php

```diff
@@ -248,243 +371,280 @@ function tsml_import_page()
-				<?php } elseif ($total = count($meetings)) { ?>
-					<div id="tsml_import_progress" class="progress" data-total="<?php echo $total ?>">
-						<div class="progress-bar"></div>
-					</div>
-					<ol id="tsml_import_errors" class="error inline hidden"></ol>
-				<?php } ?>
-
-				<!-- Import Data Sources -->
-				<div class="postbox stack">
-					<h2><?php _e('Import Data Sources', '12-step-meeting-list') ?></h2>
-					<p>
-						<?php printf(__('Data sources are JSON feeds that contain a website\'s public meeting data. They can be used to aggregate meetings from different sites into a single master list. 
-				Data sources listed below will pull meeting information into this website. A configurable schedule allows for each enabled data source to be scanned at least once per day looking 
-				for updates to the listing. Change Notification email addresses are sent an email when action is required to re-sync a data source with its meeting list information. 
-				Please note: records that you intend to maintain on your website should always be imported using the Import CSV feature below. <b>Data Source records will be overwritten when the 
-				parent data source is refreshed.</b> More information is available at the <a href="%s" target="_blank">Meeting Guide API Specification</a>.', '12-step-meeting-list'), 'https://github.com/code4recovery/spec') ?>
-					</p>
-					<?php if (!empty($tsml_data_sources)) { ?>
-						<table>
-							<thead>
-								<tr>
-									<th class="small align-center"></th>
-									<th><?php _e('Feed', '12-step-meeting-list') ?></th>
-									<th class="align-left"><?php _e('Parent Region', '12-step-meeting-list') ?></th>
-									<th class="align-left"><?php _e('Change Detection', '12-step-meeting-list') ?></th>
-									<th class="align-center"><?php _e('Meetings', '12-step-meeting-list') ?></th>
-									<th class="align-right"><?php _e('Last Refresh', '12-step-meeting-list') ?></th>
-									<th class="small"></th>
-								</tr>
-							</thead>
-							<tbody>
-								<?php foreach ($tsml_data_sources as $feed => $properties) { ?>
-									<tr data-source="<?php echo $feed ?>">
-										<td class="small ">
-											<form method="post" action="<?php echo $_SERVER['REQUEST_URI'] ?>">
-												<?php wp_nonce_field($tsml_nonce, 'tsml_nonce', false) ?>
-												<input type="hidden" name="tsml_add_data_source" value="<?php echo $feed ?>">
-												<input type="hidden" name="tsml_add_data_source_name" value="<?php echo @$properties['name'] ?>">
-												<input type="hidden" name="tsml_add_data_source_parent_region_id" value="<?php echo @$properties['parent_region_id'] ?>">
-												<input type="hidden" name="tsml_add_data_source_change_detect" value="<?php echo @$properties['change_detect'] ?>">
-												<input type="submit" value="Refresh" class="button button-small">
-											</form>
-										</td>
-										<td>
-											<a href="<?php echo $feed ?>" target="_blank">
```

Comment:
Also, just a cosmetic change. The link for CSV doesn't (and presumably shouldn't) open the file.

```
<?php if ($properties['type'] !== 'CSV') { ?>
	<a href="<?php echo $feed ?>" target="_blank">
		<?php echo !empty($properties['name']) ? $properties['name'] : __('Unnamed Feed', '12-step-meeting-list') ?>
	</a>
<?php } else { ?>
	<?php echo ! empty( $properties['name'] ) ? $properties['name'] : __( 'Unnamed Feed', '12-step-meeting-list' ); ?>
<?php } ?>
```

Reply:
I'm not completely comfortable with this code. I've never seen it executed, and I haven't done much in the way of trying to load anything but CSV files.

Reply:
When I add a CSV and it shows up in the feed list, currently the feed name is a link. When clicked, it opens a new browser window with just "meeetings.csv" in the address bar and of course that throws a 404. I looked to see if you were saving the uploaded csv on the server somewhere (like in wp-content/uploads) but couldn't find the file so I assume you're removing it after processing? I just figured removing the link for csv uploads would be the way to go, but it you want people to be able to open the csv and review it, then this isn't the way to go.